application/src/main/java/org/togetherjava/tjbot/commands/system/BotCore.java

```diff
@@ -290,6 +308,40 @@ public void onUserContextInteraction(final UserContextInteractionEvent event) {
                 UserContextCommand.class).onUserContext(event));
     }
 
+    /**
+     * Returns the {@link ComponentId} of the given componentId. If the component has expired, the
+     * user gets a message stating that.
+     *
+     * @param event the {@link IReplyCallback event} to reply to
+     * @param componentId the component's ID
+     *
+     * @return the relating {@link ComponentId}
+     */
+    @Nullable
+    public ComponentId handleGetComponentId(IReplyCallback event, String componentId) {
+        Optional<ComponentId> componentIdOpt;
+        try {
+            componentIdOpt = componentIdParser.parse(componentId);
+        } catch (InvalidComponentIdFormatException e) {
+            logger
+                .error("Unable to route event (#{}) back to its corresponding user interactor. The component ID was in an unexpected format."
+                        + " All button and menu events have to use a component ID created in a specific format"
+                        + " (refer to the documentation of UserInteractor). Component ID was: {}",
+                        event.getId(), componentId, e);
```

Comment:
this log would be more fitting if it actually says modal instead of "button and menu events". it has the context that its a modal, so it can also say that